docs/user-guide/operations/configuration.md

```diff
@@ -338,3 +338,44 @@ datanode_lease_secs = 30
 | server_addr         | String  | The communication server address for frontend and datanode to connect to metasrv,  "127.0.0.1:3002" by default for localhost |
 | store_addr          | String  | Etcd server address, "127.0.0.1:2379" by default                                                                        |
 | datanode_lease_secs | Integer | Datanode lease in seconds, 15 seconds by default.                                                                       |
+
+## Layered configuration
+
+GreptimeDB supports layered configuration and uses the following precedence order(each item takes precedence over the item below it):
+
+- Command line flags
+- Configuration file
+- Environment variables
+- Default values
+
+Every item in the configuration file can be mapped into environment variables. For example, if we want to set the configuration item `max_inflight_tasks ` of datanode by environment variables:
+
+```toml
+# ...
+[storage.compaction]
+# Max task number that can concurrently run.
+max_inflight_tasks = 4
+# ...
+```
+
+You can use the following format:
```

Comment:
```suggestion
You can also use the following format in your own environment file:
```

I guess this is for users' own environment file, is it right?

Reply:
@nicecui The current implementation doesn't read environment variables from the file. We can use the following shell command to setup the environment variables for example:

```shell
$ export GREPTIMEDB_DATANODE__STORAGE__COMPACTION__MAX_INFLIGHT_TASKS=4
$ echo $GREPTIMEDB_DATANODE__STORAGE__COMPACTION__MAX_INFLIGHT_TASKS
4
```

I already update the docs to add more details.